sdk/communication/Azure.Communication.Administration/samples/Sample2_PhoneNumberAdministrationClient.md

```diff
@@ -0,0 +1,214 @@
+```
+
+## Get area codes
+
+Fetching area codes for geographic phone plans will require the the location options queries set. You must include the chain of geographic locations traversing down the location options object returned by the GetPhonePlanLocationOptions.
+
+```C# Snippet:GeographicalAreaCodes
+var locationOptionsResponse = client.GetPhonePlanLocationOptions(countryCode, geographicPhonePlanGroupId, geographicPhonePlanId);
+var state = locationOptionsResponse.Value.LocationOptions.Options.First();
+
+var locationOptionsQueries = new List<LocationOptionsQuery>{
+    new LocationOptionsQuery
+    {
+        LabelId = "state",
+        OptionsValue = state.Value
+    },
+    new LocationOptionsQuery
+    {
+        LabelId = "city",
+        OptionsValue = state.LocationOptions.First().Options.First().Value
+    }
+};
+
+var areaCodes = client.GetAllAreaCodes(geographicPhonePlan.LocationType.ToString(), countryCode, geographicPhonePlan.PhonePlanId, locationOptionsQueries);
+
+foreach (var areaCode in areaCodes.Value.PrimaryAreaCodes)
+{
+    Console.WriteLine($"Primary area code: {areaCode}");
+}
+```
+
+Area codes for toll free phone plans can be found in the plan.
+
+```C# Snippet:TollFreePlanAreCodes
+var phonePlans = client.GetPhonePlans(countryCode, tollFreePhonePlanGroupId, locale);
+var tollFreePhonePlan = phonePlans.First();
+
+foreach (var areaCode in tollFreePhonePlan.AreaCodes)
+{
+    Console.WriteLine($"Area code: {areaCode}");
+}
+```
+
+## Reserve phone numbers
+
+Phone numbers need to be reserved for purchase. Reservation is a long running operation that can be started by CreateReservationOptions function that returns an PhoneNumberReservationOperation object. PhoneNumberReservationOperation can be used to update status of the operation and to check for completeness.
+
+```C# Snippet:ReservePhoneNumbers
+var reservationName = "My reservation";
+var reservationDescription = "reservation description";
+var reservationOptions = new CreateReservationOptions(reservationName, reservationDescription, new[] { phonePlanId }, areaCode);
+reservationOptions.Quantity = 1;
+
+var reservationOperation = client.StartReservation(reservationOptions);
+
+while (true)
+{
+    reservationOperation.UpdateStatus();
+    if (reservationOperation.HasCompleted)
+    {
+        break;
+    }
+
+    Thread.Sleep(1000);
+}
+```
+
+## Persist reserve phone numbers operation
+
+You can persist phone number reservation operation Id so that you can get back later to check operation status.
+
+```C# Snippet:PersistReservePhoneNumbersOperation
+var reservationId = reservationOperation.Id;
+
+// persist reservationOperationId and then continue with new operation
+
+var newPhoneNumberReservationOperation = new PhoneNumberReservationOperation(client, reservationId);
+
+while (true)
+{
+    newPhoneNumberReservationOperation.UpdateStatus();
+    if (newPhoneNumberReservationOperation.HasCompleted)
+    {
+        break;
+    }
+
+    Thread.Sleep(1000);
```

Comment:
@christothes is there a better recommended way to have sync tests for long running operations?

Reply:
@christothes @RezaJooyandeh just a reminder that this test code is copied over MD sample file

Reply:
Let's follow how KeyVault has written the code. https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/keyvault/Azure.Security.KeyVault.Keys/samples/Sample1_HelloWorld.md#purging-a-deleted-key

```csharp
while (!newPhoneNumberReservationOperation.HasCompleted)
{
    Thread.Sleep(2000);

    operation.UpdateStatus();
}
```

Reply:
Done